azurerm/internal/services/apimanagement/api_management_gateway_data_source.go

```diff
@@ -0,0 +1,87 @@
+package apimanagement
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/apimanagement/parse"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/services/apimanagement/schemaz"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tf/pluginsdk"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceApiManagementGateway() *pluginsdk.Resource {
+	return &pluginsdk.Resource{
+		Read: dataSourceApiManagementGatewayRead,
+
+		Timeouts: &pluginsdk.ResourceTimeout{
+			Read: pluginsdk.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*pluginsdk.Schema{
+			"name": schemaz.SchemaApiManagementChildDataSourceName(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"api_management_name": schemaz.SchemaApiManagementDataSourceName(),
+
+			"description": {
+				Type:     pluginsdk.TypeString,
+				Computed: true,
+			},
+
+			"location": {
+				Type:     pluginsdk.TypeList,
+				Computed: true,
+				Elem: &pluginsdk.Resource{
+					Schema: map[string]*pluginsdk.Schema{
+						"name": {
+							Type:     pluginsdk.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func dataSourceApiManagementGatewayRead(d *pluginsdk.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).ApiManagement.GatewayClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	resourceGroup := d.Get("resource_group_name").(string)
+	serviceName := d.Get("api_management_name").(string)
+	name := d.Get("name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, serviceName, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("make read request gateway %q (Resource Group %q / API Management Service %q) was not found", name, resourceGroup, serviceName)
+		}
+
+		return fmt.Errorf("making read request for gateway %q (Resource Group %q / API Management Service %q): %+v", name, resourceGroup, serviceName, err)
+	}
```

Comment:
We can then use the ID struct's fields/methods here

```suggestion
	resp, err := client.Get(ctx, id.ResourceGroup, id.ServiceName, id.Name)
	if err != nil {
		if utils.ResponseWasNotFound(resp.Response) {
			return fmt.Errorf("%s was not found", id)
		}

		return fmt.Errorf("making read request for %s: %+v", id, err)
	}
```